inst/include/interface/rcpp/rcpp_objects/rcpp_fleet.hpp

```diff
@@ -31,4 +31,203 @@
 #ifndef FIMS_INTERFACE_RCPP_RCPP_OBJECTS_RCPP_FLEET_HPP
 #define FIMS_INTERFACE_RCPP_RCPP_OBJECTS_RCPP_FLEET_HPP
 
+#include "../../../population_dynamics/fleet/fleet.hpp"
+#include "rcpp_interface_base.hpp"
+
+/**
+ * @brief Rcpp interface for Fleet as an S4 object. To instantiate
+ * from R:
+ * fleet <- new(fims$Fleet)
+ * 
+ */
+class FleetInterface : public FIMSRcppInterfaceBase
+{
+    int observed_index_data_id = -999; /*!< id of observed index data object*/
+    int observed_agecomp_data_id = -999; /*!< id of observed agecomp data object*/
+    int index_likelihood_id = -999; /*!< id of index likelihood component*/
+    int agecomp_likelihood_id = -999; /*!< id of agecomp likelihood component*/
```

Comment:
Code for setting up 0-3rd order derivative fleet model components
appears flawless. I wonder if it would be useful to order the definition
of the int *_id objects (lines 45-49) in the same order as their treatment
in the void functions below?